src/theory/bv/theory_bv_rewrite_rules_normalization.h

```diff
@@ -408,31 +390,62 @@ Node RewriteRule<MultSimplify>::apply(TNode node) {
   unsigned size = utils::getSize(node); 
   BitVector constant(size, Integer(1));
 
-  std::vector<Node> children; 
-  for(unsigned i = 0; i < node.getNumChildren(); ++i) {
-    TNode current = node[i];
+  bool isNeg = false;
+  std::vector<Node> children;
+  for (const TNode& current : node)
+  {
     if (current.getKind() == kind::CONST_BITVECTOR) {
       BitVector value = current.getConst<BitVector>();
       constant = constant * value;
       if(constant == BitVector(size, (unsigned) 0)) {
         return utils::mkConst(size, 0); 
       }
+    }
+    else if (current.getKind() == kind::BITVECTOR_NEG)
+    {
+      isNeg = !isNeg;
+      children.push_back(current[0]);
     } else {
       children.push_back(current); 
     }
   }
+  BitVector oValue = BitVector(size, static_cast<unsigned>(1));
+  Node negOne = utils::mkConst(-oValue);
+  BitVector noValue = negOne.getConst<BitVector>();
+
+  if (children.empty())
+  {
+    if (isNeg)
```

Comment:
This can't happen. As soon as `isNeg` is set to true, `children.size() > 0`. Further, negated constants are always rewritten to a constant, so `isNeg` should be false here anyways.